### PR TITLE
Case value when value then

### PR DESCRIPTION
**Pull request by galsalomon66**

### CODE REVIEW

**Comment by albin-antony** [comment]
Merged the commit in #49 . Thank you @galsalomon66

File: include/s3select_functions.h [hunk 1]
```diff
@@ -58,6 +58,7 @@ enum class s3select_func_En_t {ADD,
                                VERSION,
                                CASE_WHEN_ELSE,
                                WHEN_THAN,
+                               WHEN_THEN_SIMPLE,
```
Comment (galsalomon66) [inline include/s3select_functions.h:61]: nit
replace that with WHEN_VALUE_THEN 
simple is confusing 

better to use self-explained text when possible.

File: include/s3select_functions.h [hunk 2]
```diff
@@ -102,6 +103,7 @@ class s3select_functions
     {"#like_predicate#", s3select_func_En_t::LIKE},
     {"version", s3select_func_En_t::VERSION},
     {"#when-than#", s3select_func_En_t::WHEN_THAN},
+    {"#when-then-simple#", s3select_func_En_t::WHEN_THEN_SIMPLE},
```
Comment (galsalomon66) [inline include/s3select_functions.h:106]: replace /simple/

File: include/s3select_functions.h [hunk 3]
```diff
@@ -1315,6 +1317,39 @@ struct _fn_when_than : public base_function {
   }
 };
 
+struct _fn_when_then_simple : public base_function {
```
Comment (galsalomon66) [inline include/s3select_functions.h:1320]: replace /simple/

**Comment by galsalomon66** [comment]
need to add complex and nested queries
for one example queries with mix of case-when-condition and case-value-when

**Pull request closed without merging**